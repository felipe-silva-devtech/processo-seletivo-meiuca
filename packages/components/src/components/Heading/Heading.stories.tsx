import React from "react";

import { StoryFn, Meta } from "@storybook/react";

import { Heading, HeadingProps } from ".";

export default {
  title: "Components / Heading",
  component: Heading,
  argTypes: {
    content: { control: { type: "text" } },
  },
  args: {
    content: "Heading SM",
  },
} as Meta<HeadingProps>;

export const Default: StoryFn<typeof Heading> = (args) => <Heading {...args} />;
