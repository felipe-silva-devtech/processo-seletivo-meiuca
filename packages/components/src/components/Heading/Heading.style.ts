import styled from "@emotion/styled";
import { basicProps } from "../../utils/styled-system-utils";

export const StyledHeading = styled("h1")`
  ${basicProps}

  font-size: ${({ theme }) => theme.fontSizes.lg};
  line-height: ${({ theme }) => theme.lineHeights.distant};
  font-family: ${({ theme }) => theme.fonts.highlight};
  font-weight: ${({ theme }) => theme.fontWeights.bold};
  color: ${({ theme }) => theme.colors.neutral["01"]};
`;
