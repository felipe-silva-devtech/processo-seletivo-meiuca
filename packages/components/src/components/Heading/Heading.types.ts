import React from "react";

export type HeadingProps = React.PropsWithChildren & {
  content?: string;
};
