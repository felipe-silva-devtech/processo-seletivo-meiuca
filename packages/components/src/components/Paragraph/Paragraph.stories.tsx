import React from "react";

import { StoryFn, Meta } from "@storybook/react";

import { Paragraph, ParagraphProps } from ".";

export default {
  title: "Components / Paragraph",
  component: Paragraph,
  argTypes: {
    content: { control: { type: "text" } },
  },
  args: {
    content:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. " +
      "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, " +
      "when an unknown printer took a galley of type and scrambled it to make a type specimen book. " +
      "It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged.",
  },
} as Meta<ParagraphProps>;

export const Default: StoryFn<ParagraphProps> = (args) => (
  <Paragraph {...args} />
);
