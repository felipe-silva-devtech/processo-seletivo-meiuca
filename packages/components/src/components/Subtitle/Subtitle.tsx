import React from "react";

import { StyledSubtitle } from "./Subtitle.style";
import { SubtitleProps } from "./Subtitle.types";

export const Subtitle: React.FC<SubtitleProps> = ({
  children,
  content = "",
  ...props
}) => {
  return (
    <StyledSubtitle {...props}>{children ? children : content}</StyledSubtitle>
  );
};
