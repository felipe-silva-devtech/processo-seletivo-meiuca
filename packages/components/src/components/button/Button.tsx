import React from "react";

import { StyledButton } from "./Button.style";
import { ButtonProps } from "./Button.types";

export const Button: React.FC<ButtonProps> = ({
  children,
  label,
  ...props
}) => {
  return (
    <StyledButton type="button" bg="brand.primary.01" {...props}>
      {children ? children : label}
    </StyledButton>
  );
};
