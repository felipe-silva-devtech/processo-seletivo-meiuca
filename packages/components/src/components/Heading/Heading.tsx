import React from "react";

import { StyledHeading } from "./Heading.style";
import { HeadingProps } from "./Heading.types";

export const Heading: React.FC<HeadingProps> = ({
  children,
  content = "",
  ...props
}) => {
  return (
    <StyledHeading {...props}>{children ? children : content}</StyledHeading>
  );
};
