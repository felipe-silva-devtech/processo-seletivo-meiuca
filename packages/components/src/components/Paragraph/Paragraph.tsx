import React from "react";

import { StyledParagraph } from "./Paragraph.style";

export const Paragraph = ({ children, content = "", ...props }) => {
  return (
    <StyledParagraph {...props}>
      {children ? children : content}
    </StyledParagraph>
  );
};
