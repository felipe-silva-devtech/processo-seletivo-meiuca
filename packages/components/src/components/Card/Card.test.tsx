import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

import { ThemeProvider, theme } from "@psm/theme";
import { Card } from ".";
import { Heading } from "../Heading";
import { Subtitle } from "../Subtitle";
import { Paragraph } from "../Paragraph";
import { Button } from "../Button";

function renderWithWrapper(
  component?: React.ReactElement,
  options?: RenderOptions
) {
  const Wrapper = <ThemeProvider theme={theme}>{component}</ThemeProvider>;

  return render(Wrapper, options);
}

describe("Card", () => {
  const headingContent = "Heading SM";
  const subtitleContent = "Subtitle SM";
  const paragraphContent =
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry.";
  const buttonLabel = "Button Label";

  it("render component", () => {
    const { container } = renderWithWrapper(
      <Card
        heading={headingContent}
        subtitle={subtitleContent}
        paragraph={paragraphContent}
        buttonLabel={buttonLabel}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("render component with children", () => {
    const { container } = renderWithWrapper(
      <Card>
        <Heading>{headingContent}</Heading>
        <Subtitle>{subtitleContent}</Subtitle>
        <Paragraph>{paragraphContent}</Paragraph>
        <Button>{buttonLabel}</Button>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });

  it("render component without subtitle", () => {
    const { container } = renderWithWrapper(
      <Card>
        <Heading>{headingContent}</Heading>
        <Paragraph>{paragraphContent}</Paragraph>
        <Button>{buttonLabel}</Button>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });
});
