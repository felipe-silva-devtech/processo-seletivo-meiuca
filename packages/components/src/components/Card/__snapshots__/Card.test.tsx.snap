// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render component 1`] = `
<div>
  <div
    class="css-17tpkix"
    width="357px"
  >
    <h1
      class="css-1uol47"
    >
      Heading SM
    </h1>
    <h2
      class="css-1g1spkf"
    >
      Subtitle SM
    </h2>
    <p
      class="css-1sclvem"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </p>
    <button
      class="css-gkv5w0"
    >
      Button Label
    </button>
  </div>
</div>
`;

exports[`Card render component with children 1`] = `
<div>
  <div
    class="css-17tpkix"
    width="357px"
  >
    <h1
      class="css-1j1wjfe"
    >
      Heading SM
    </h1>
    <h2
      class="css-9rgr1i"
    >
      Subtitle SM
    </h2>
    <p
      class="css-2wh3l1"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </p>
    <button
      class="css-gkv5w0"
    >
      Button Label
    </button>
  </div>
</div>
`;

exports[`Card render component without subtitle 1`] = `
<div>
  <div
    class="css-17tpkix"
    width="357px"
  >
    <h1
      class="css-1j1wjfe"
    >
      Heading SM
    </h1>
    <p
      class="css-2wh3l1"
    >
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
    </p>
    <button
      class="css-gkv5w0"
    >
      Button Label
    </button>
  </div>
</div>
`;
